### PR TITLE
add azure prefix in template to avoid clash across providers

### LINKED ISSUE #436: AZURE_ prefixes are required in templates
/kind bug

What steps did you take and what happened:
[A clear and concise description of what the bug is.]
CONTROL_PLANE_MACHINE_TYPE and NODE_MACHINE_TYPE template variables should have an AZURE_ prefix. Currently have a clash in when using ~/.cluster-api/clusterctl.yaml across providers

What did you expect to happen:
Can use Azure and AWS providers simultaneously.

Anything else you would like to add:
[Miscellaneous information that will assist in solving the issue.]

I will be updating quick start docs to state the following:

```bash
# Name of the resource group to provision into
$ export AZURE_RESOURCE_GROUP="kubernetesResourceGroup"
# Name of the Azure datacenter location
$ export AZURE_LOCATION="centralus"
# Name of the virtual network in which to provision the cluster
$ export AZURE_VNET_NAME="kubernetesNetwork"
# Select machine types
$ export AZURE_CONTROL_PLANE_MACHINE_TYPE="Standard_D4a_v4"
$ export AZURE_NODE_MACHINE_TYPE="Standard_D4a_v4"
$ A SSH key to use for break-glass access
$ export AZURE_SSH_PUBLIC_KEY="ssh-rsa AAAAB3N..."
```

For more information about authorization, AAD, or requirements for Azure, visit the [Azure provider prerequisites] document.

Related to https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1631

### PR DESCRIPTION
**Pull request by nader-ziada**
**What this PR does / why we need it**:
add azure prefix in template to avoid clash across providers

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #436 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/resource_generators.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func cloudConfig(clusterName string, creds auth.Creds) string {
 }
 
 func sshkey() (string, error) {
-	// TODO Load from AZURE_SSH_PUBLIC_KEY_FILE if set
+	// TODO Load from AZURE_AZURE_SSH_PUBLIC_KEY_FILE if set
```
Comment (detiber) [inline test/e2e/resource_generators.go]:
```suggestion
	// TODO Load from AZURE_SSH_PUBLIC_KEY_FILE if set
```

File: docs/development.md [hunk 2]
```diff
@@ -153,18 +153,18 @@ export AZURE_CLIENT_SECRET_B64="$(echo -n "$AZURE_CLIENT_SECRET" | base64 | tr -
...
 
 # Generate SSH key.
-# If you want to provide your own key, skip this step and set SSH_PUBLIC_KEY to your existing file.
+# If you want to provide your own key, skip this step and set AZURE_SSH_PUBLIC_KEY to your existing file.
 SSH_KEY_FILE=.sshkey
 rm -f "${SSH_KEY_FILE}" 2>/dev/null
 ssh-keygen -t rsa -b 2048 -f "${SSH_KEY_FILE}" -N '' 1>/dev/null
 echo "Machine SSH key generated in ${SSH_KEY_FILE}"
-export SSH_PUBLIC_KEY=$(cat "${SSH_KEY_FILE}.pub" | base64 | tr -d '\r\n')
+export AZURE_SSH_PUBLIC_KEY=$(cat "${SSH_KEY_FILE}.pub" | base64 | tr -d '\r\n')
```
Comment (awesomenix) [inline docs/development.md]: Shouldnt this be the same across all providers? Seems like WORKER_MACHINE_COUNT and KUBERNETES_VERSION remains intact.

Comment (nader-ziada) [reply docs/development.md]: I assumed you could have different keys across providers

File: docs/development.md [hunk 3]
```diff
@@ -141,7 +141,7 @@ Here is a list of commonly overriden configuration parameters (the full list is
 ```bash
 # Cluster settings.
 export CLUSTER_NAME="capz-cluster"
```
Comment (CecileRobertMichon) [inline docs/development.md]: should this be 
```suggestion
export AZURE_CLUSTER_NAME="capz-cluster"
```
?

Comment (nader-ziada) [reply docs/development.md]: not sure, thought about it, but wasn't sure if we should change it

Comment (detiber) [reply docs/development.md]: I do believe CLUSTER_NAME should likely be common to all providers.

Comment (CecileRobertMichon) [reply docs/development.md]: discussed offline, let's keep this one common

**Pull request merged**